### ISSUE TITLE
Mutations don't update the cache unless an existing query has been sent

### ISSUE THREAD

**Issue by DustinJSilk** [FormidableLabs/urql#2750]
### Describe the bug

Documents that are returned from a mutation with an "id" and "__typename" field should be cached. Currently, mutation results are only cached if a query has been sent or subscribed to first.

The reproduction link uses the React library, however, this is experienced using just the core library too.

### Reproduction

https://stackblitz.com/edit/react-ts-yqn9jf?devToolsHeight=33&file=App.tsx

### Urql version

@urql/core: 3.0.3
@urql/exchange-graphcache: 5.0.2
graphql: 16.6.0

### Validations

- [X] I can confirm that this is a bug report, and not a feature request, RFC, question, or discussion, for which [GitHub Discussions](https://github.com/FormidableLabs/urql/discussions) should be used
- [X] Read the [docs](https://formidable.com/open-source/urql/docs/).
- [X] Follow our [Code of Conduct](https://github.com/FormidableLabs/urql/blob/main/CODE_OF_CONDUCT.md)

**Comment by JoviDeCroock** [comment]
Mutations aren't meant to be cached, you can however create an updater to link this. Even if we would cache Film:1 in this scenario it would get garbage collected so we need to link it up to the Query field. i.e. you can write an updater that does the following

```js
cache.link('Query', 'film', { id }, { id, __typename: 'Film' })
```

We basically delete all entities with no relation to the root-entity else the cache would just keep growing and consuming memory

**Comment by DustinJSilk** [comment]
Oh i didnt realise it wasnt meant to be cached! I'll read through this article, thanks for sharing!

The only reason this is an issue for me is because i'm starting subscriptions during SSR and resuming them on the client but because I dont need all the data during hydration i'm not using the SSR module, which would solve this problem for me. 

I might dig into the SSR module and fork it for my use case so it only stores the data I actually need returned after SSR.

I added some more info in this [discussion](https://github.com/FormidableLabs/urql/discussions/2752) that I opened up earlier which helps to explain this some more.

Thanks @JoviDeCroock
